src/main/java/com/example/chat/groupchatbackend/controllers/UserController.java

```diff
@@ -36,6 +45,59 @@ public ResponseEntity register(@RequestBody User user) {
                 .body("User already exists");
     }
 
+    @GetMapping("/messages/{conversationName}?newerThan={timestamp}")
+    public List<Message> getAllMessages(@PathVariable String conversationName, @PathVariable Long timestamp) {
+        if (timestamp == null) {
+            Conversation conversation = getConversation(conversationName);
+            return conversation.getMessages();
+        } else {
+            Conversation conversation = getConversation(conversationName);
+            LocalDateTime date = Instant.ofEpochMilli(timestamp).atZone(ZoneId.systemDefault()).toLocalDateTime();
+            return conversation.getMessages().stream()
+                    .filter(message -> message.getTimestamp().isAfter(date))
+                    .collect(Collectors.toList());
+        }
+    }
+
+    @PostMapping("/messages/{conversationName}")
+    public void postMessageToConversation(
+            @PathVariable("conversationName") String conversationName,
+            @RequestBody String text) {
+        if (conversationsRepository.findByName(conversationName).isPresent()) {
+            Message message = new Message(1, text, LocalDateTime.now());
+            messagesRepository.save(message);
+            Conversation conversation = conversationsRepository.findByName(conversationName).get();
+            conversation.getMessages().add(message);
+            conversationsRepository.save(conversation);
+        } else {
+            throw new EntityNotFoundException("Conversation with such name doesn't exists");
+        }
+    }
+
+    private Conversation getConversation(@PathVariable String conversationName) {
+        return conversationsRepository.findByName(conversationName).orElseThrow(() -> new RuntimeException());
+    }
+
+    @GetMapping("/messages")
+    public Iterable<Message> getAllMessages(){
+        return messagesRepository.findAll();
+    }
+
+    @PutMapping("/messages/{id}")
+    @Transactional
+    public Message editMessage(@PathVariable int id, @RequestBody String text){
+        Message message = messagesRepository.findById(id).orElseThrow(() -> new RuntimeException("message doesn't exist"));
+        message.setText(text);
+        return message;
+    }
+
+    @DeleteMapping("/messages/{id}")
+    public ResponseEntity deleteMessage(@PathVariable int id){
+        Message message = messagesRepository.findById(id).orElseThrow(() -> new RuntimeException("message doesn't exist"));
```

Comment:
this line is identical to one from Put method, let's create a new method (getMessageById) and use it in both places